server/src/main/java/org/elasticsearch/search/aggregations/bucket/histogram/InternalAutoDateHistogram.java

```diff
@@ -238,6 +237,22 @@ protected void doWriteTo(StreamOutput out) throws IOException {
         out.writeVInt(targetBuckets);
     }
 
+    public DateHistogramInterval getInterval() {
+        RoundingInfo roundingInfo = this.bucketInfo.roundingInfos[this.bucketInfo.roundingIdx];
+        String unitAbbreviation = roundingInfo.unitAbbreviation;
+        if (buckets.size() <= 1) {
+            int innerInterval = roundingInfo.innerIntervals[0];
+            return new DateHistogramInterval(Integer.toString(innerInterval) + unitAbbreviation);
+        }
+        long intervalInMillis = buckets.get(1).key - buckets.get(0).key;
+        for (int interval : roundingInfo.innerIntervals) {
+            if (roundingInfo.getRoughEstimateDurationMillis() * interval == intervalInMillis) {
+                return new DateHistogramInterval(Integer.toString(interval) + unitAbbreviation);
+            }
+        }
+        return new DateHistogramInterval(Integer.toString(roundingInfo.innerIntervals[0]) + unitAbbreviation);
+    }
```

Comment:
I'm not sure we should calculate the interval here as we could end up with bugs where this is different from the actual interval used. Instead could we have a field for the interval store in this class which is populated int he `AutoDateHistogramAggregator.buildAggregation()` and also in `doReduce()` when we build the new instance following a reduce? I think we will have all the information we need in both those places?